app/assets/javascripts/discourse/tests/acceptance/user-menu-test.js

```diff
@@ -748,6 +748,34 @@ acceptance("User menu", function (needs) {
       await click("#site-logo");
     }
   });
+
+  test("tabs have hrefs and can be opened in new window/tab", async function (assert) {
+    await visit("/");
+    await click(".d-header-icons .current-user");
+
+    assert
+      .dom("#user-menu-button-replies")
+      .hasAttribute("href", "/u/eviltrout/notifications/responses");
+
+    // Add a top-level click listener to stub attempts to open a new window/tab
+    const newWindowOpenedAssertion = assert.async();
+    const interceptor = (event) => {
+      event.preventDefault();
+
+      newWindowOpenedAssertion();
+      const target = event.target;
+      assert.strictEqual(target.tagName, "A");
+      assert.true(target.href.endsWith("/u/eviltrout/notifications/responses"));
+    };
+
+    window.addEventListener("click", interceptor);
+
+    try {
+      await click("#user-menu-button-replies", { shiftKey: true });
```

Comment:
I'm not sure here but in the acceptance test environment, does `await`-ing on `click`, waits for all event listeners to complete?

Reply:
I don't think it waits for event listeners, but we use the `assert.async()` function here which, from my understanding of the [docs](https://api.qunitjs.com/assert/async/), doesn't let the test case finish until the callback returned by the function is invoked. (David please correct me if I got this wrong 😅)

Reply:
Yep exactly - `assert.async()` returns a callback function, and the test will automatically wait until that callback is triggered. 👌